odl/tomo/operators/xray_trafo.py

```diff
@@ -176,57 +183,145 @@ def _call(self, inp):
     @property
     def adjoint(self):
         """Return the adjoint operator."""
-        return self._adjoint
-
+        return XrayBackProjector(self.domain, self.geometry, self.backend,
+                                 **self.kwargs)
 
-class DiscreteXrayTransformAdjoint(Operator):
 
-    """The adjoint of the discrete X-ray transform."""
+class XrayBackProjector(Operator):
+    """The adjoint of the discrete X-ray transform between `L^p` spaces."""
 
-    def __init__(self, forward):
+    def __init__(self, discr_range, geometry, backend='astra', **kwargs):
         """Initialize a new instance.
 
         Parameters
         ----------
-        forward : `DiscreteXrayTransform`
-            An instance of the discrete X-ray transform
+        discr_range : `odl.DiscreteLp`
+            Reconstruction space, the range of the back-projector
+        geometry : `Geometry`
+            The geometry of the transform, contains information about
+            the operator domain. It needs to have a sampling grid for
+            motion and detector parameters.
+        backend : {'astra', 'astra_cuda', 'astra_cpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+            'astra': ASTRA toolbox, uses CPU or CUDA depending on the
+            underlying data space of ``discr_range``
```

Comment:
I think this was because of the intention to decide on to use cpu or cuda depending on the provided space and because I've anticipated, back then when Willem-Jan was here, that we soon will have an ASTRA version which can use a shared CUDA memory. But, since we don't when this will happen, I'm gonna remove the 'astra' option. Or what do you think?